homeassistant/components/nfandroidtv/notify.py

```diff
@@ -132,21 +132,39 @@ def send_message(self, message: str, **kwargs: Any) -> None:
                         "Invalid interrupt-value: %s", data.get(ATTR_INTERRUPT)
                     )
             if imagedata := data.get(ATTR_IMAGE):
-                image_file = self.load_file(
-                    url=imagedata.get(ATTR_IMAGE_URL),
-                    local_path=imagedata.get(ATTR_IMAGE_PATH),
-                    username=imagedata.get(ATTR_IMAGE_USERNAME),
-                    password=imagedata.get(ATTR_IMAGE_PASSWORD),
-                    auth=imagedata.get(ATTR_IMAGE_AUTH),
-                )
+                if isinstance(imagedata, str):
+                    image_file = (
+                        self.load_file(url=imagedata)
+                        if imagedata.startswith("http")
+                        else self.load_file(local_path=imagedata)
+                    )
+                elif isinstance(imagedata, dict):
+                    image_file = self.load_file(
+                        url=imagedata.get(ATTR_IMAGE_URL),
+                        local_path=imagedata.get(ATTR_IMAGE_PATH),
+                        username=imagedata.get(ATTR_IMAGE_USERNAME),
+                        password=imagedata.get(ATTR_IMAGE_PASSWORD),
+                        auth=imagedata.get(ATTR_IMAGE_AUTH),
+                    )
+                else:
+                    _LOGGER.error("Unexpected image data passed to notify service")
```

Comment:
I think this should raise a `ServiceValidationError` instead of just logging a warning, @jbouwh can you confirm?